certora/helpers/MerkleTreeLib.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+library MerkleTreeLib {
+    using MerkleTreeLib for Node;
+
+    struct Leaf {
+        address addr;
+        uint256 value;
+    }
+
+    struct InternalNode {
+        address addr;
```

Comment:
The keys of the nodes are addresses, to be able to easily make sure that there are no duplicate addresses in the tree. So internal nodes have an address, but we don't really care about them, except that we want to easily create an address that is distinct from the other ones that are currently used (that's why I use this weird hashing)

Reply:
but how having an address on internal nodes prevent having duplicates leaves?

Reply:
No we don't really care about addresses of internal nodes, but we use the same key format (an address) for both. This is because the tree is defined as a single mapping:
https://github.com/morpho-org/morpho-optimizers/blob/8ad06d22029032d0bd6fd826d213b385635466fc/certora/helpers/MerkleTreeLib.sol#L34-L37

 So once we have an address for leaves, then we have an address for internal nodes

Reply:
get it